lib/handlers/post.js

```diff
@@ -80,20 +81,19 @@ function handler (req, res, next) {
 
   function one () {
     debug('Receving one file')
-    var linkHeader = header.parseMetadataFromHeader(req.get('Link'))
-    var slug = req.get('Slug')
-    ldp.post(
-      req.hostname,
-      containerPath,
-      slug,
-      req,
-      linkHeader.isBasicContainer,
+    const { slug, link, 'content-type': contentType } = req.headers
+    const links = header.parseMetadataFromHeader(link)
+    const mimeType = contentType ? contentType.replace(/\s*;.*/, '') : ''
```

Comment:
I recommend you extract this logic into a function so that it can be tested in isolation.

Reply:
Which part, the single line or the extraction of headers in general?

Reply:
Ah I was thinking that the core of this PR was "given a request, how do we infer the mime type and extension", so I was thinking that the logic for determining `mimeType` and `extension` could live in their own functions. Not a big deal, since this is covered already in tests.